### ISSUE TITLE
error do not caught - when arg is missing for function when function is arg too

### ISSUE THREAD

**Issue by vitaly-zdanevich** [PyCQA/pylint#1317]
This code does not working - because `self.myFunc()` must receive argument `x` - but PyLint do not show this:
```
import threading

def myFunc(x):
    print('myFunc started, x: %s' % x)

class MyClass(threading.Thread):
    def __init__(self, myFunc):
        threading.Thread.__init__(self)
        self.myFunc = myFunc

    def run(self):
        # TypeError: myFunc() missing 1 required positional argument: 'x'
        self.myFunc()

MyClass(myFunc).start()
```

### pylint --version output
```
pylint 1.6.5,
astroid 1.4.9
Python 3.5.2 (default, Jul 28 2016, 21:28:00)
[GCC 4.2.1 Compatible Apple LLVM 7.3.0 (clang-703.0.31)]
```

**Comment by PCManticore** [comment]
This is a classic example of a data flow problem. Pylint does not currently have support for this. There are a bunch of other issues similar to this one, needing the same support we're currently lacking.
